test/mockProviders.ts

```diff
@@ -181,18 +199,27 @@ export class EthereumProvider {
         {
           const keyParam = '0x' + payload.params[0].data.substr(10);
 
-          result = Array.isArray(this.returnData)
-            ? this.returnData.find((e) => e.key === keyParam).value
-            : this.returnData;
+          result = this.returnData.find((e) => e.key === keyParam).value;
```

Comment:
can `this.returnData.find((e) => e.key === keyParam)` ever return undefined?

Reply:
Good catch, yes it potentially can.
So far we only use it with a set of data we know will always be true as the mock input and "expected return data" come from same object. But adding a `?` can't hurt, I'll update thanks.